enum.rb

```diff
@@ -0,0 +1,521 @@
+               else
+                 !(pat === self[index])
+        end
+    end
+    temp
+  end
+
+  # rubocop: enable Style/CaseEquality, Layout/EndAlignment
+  # *********************************
+  # End of my_none? method.
+  # *********************************
+
+  # *************************(********
+  # Beginning of my_count method.
+  # *********************************
+  # rubocop: disable Style/CaseEquality
+  NOTHING = Object.new
+
+  def my_count(arg = NOTHING)
+    if block_given?
+      my_select { |index| yield index }.length
+    else
+      my_select { |index| index == arg || NOTHING === arg }.length
+    end
+  end
+
+  # rubocop: enable Style/CaseEquality
+
+  # *********************************
+  # End of my_count method.
+  # *********************************
+
+  # *********************************
+  # Beginning of my_map method.
+  # *********************************
+  def my_map
+    mapped = []
+    return to_enum :my_map, print(' No block was given.') unless block_given?
+
+    0.upto(length - 1) do |index|
+      mapped.push(yield self[index])
+    end
+    mapped
+  end
+
+  # rubocop: enable, Metrics/PerceivedComplexity
+
+  # *********************************
+  # End of my_map method.
+  # *********************************
+
+  # *********************************
+  # Beginning of my_map with proc method.
+  # *********************************
+  # rubocop: disable Layout/LineLength, Style/CaseEquality
+  def my_map_proc(procs = NOTHING)
+    mapped = []
+    unless Proc === procs
+      return to_enum :my_map_proc, print('Please enter a valid proc.', "\n", "\n")
+    end
+
+    0.upto(length - 1) do |index|
+      mapped.push(procs.call(self[index]))
+    end
+    mapped
+  end
+
+  # rubocop: enable Layout/LineLength, Style/CaseEquality
+
+  # *********************************
+  # End of my_map with proc method.
+  # *********************************
+
+  # *********************************
+  # Beginning of my_map_pb method
+  # *********************************
+
+  # rubocop: disable Metrics/AbcSize, Layout/LineLength, Style/CaseEquality
+  def my_map_pb(procs = NOTHING)
+    mapped = []
+    unless Proc === procs || block_given?
+      return to_enum :my_map_pb, print('Kindly enter a valid proc or block', "\n", "\n")
+    end
+
+    0.upto(length - 1) do |index|
+      mapped.push(procs.call(self[index])) if Proc === procs
+      mapped.push(yield self[index]) if block_given? && !Proc === procs
+    end
+    mapped
+  end
+
+  # rubocop: enable Metrics/AbcSize, Layout/LineLength, Style/CaseEquality
+  # *********************************
+  # End of my_map_pb method.
+  # *********************************
+
+  # *********************************
+  # Beginning of my_inject method.
+  # *********************************
+
+  # rubocop: disable Style/MultipleComparison, Metrics/AbcSize, Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity, Metrics/MethodLength
+  def my_inject(*args)
```

Comment:
- [ ] should be identical and returns the same thing as ruby's [`inject`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-inject). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_inject.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [ ] when a symbol is specified combines each element of the collection by applying the symbol as a named method
```ruby
array = Array.new(100) { rand(0...9) }
array.my_inject(:+) == array.inject(:+) # should true
```
- [ ] combines all elements of enum by applying a binary operation, specified by a block
```ruby
operation = proc { |sum, n| sum + n } 
search = proc { |memo, word| memo.length > word.length ? memo : word } 
array = Array.new(100){rand(0...9)}
words = %w[dog door rod blade]

array.my_inject(&operation) == array.inject(&operation) # should true
actual = range.my_inject(4) { |prod, n| prod * n }
expected = range.inject(4) { |prod, n| prod * n }
actual == expected # should true
words.my_inject(&search) == words.inject(&search) # should true
```

Reply:
Tests run satisfactorily.